markets/spot-market/contracts/modules/AsyncOrderModule.sol

```diff
@@ -57,11 +58,12 @@ contract AsyncOrderModule is IAsyncOrderModule {
             amountEscrowed = amountProvided;
         } else if (orderType == Transaction.Type.ASYNC_SELL) {
             // Get the dollar value of the provided synths
-            uint256 usdAmount = Price.synthUsdExchangeRate(
+            uint256 currentPrice = Price.getCurrentPrice(
```

Comment:
May I suggest adding `currentPrice` and `usdAmount` to the `OrderCommitted` event, because this data is transient and (almost) impossible to recover retrospectively.

Reply:
This comment is valid for all other places in the code where we rely on current prices and then do mutations. Events should include that price at the time of mutation.

Reply:
this price is valid for selling and it's on commitment, and only used for validation.  on `OrderSettled`, we do emit the price it was settled at which is far more accurate.